internal/node/process_mgr.go

```diff
@@ -9,34 +11,45 @@ import (
 // (e.g. firecracker) is employed. Note that agent processes are created asynchronously -before- any
 // workloads are deployed to them, so a workload manager can never explicitly tell a process manager
 // to create an individual process
-type ProcessManager interface {
-	// Starts the process manager. This allows the process manager to create a pool of agents however it sees fit
-	Start(ProcessSubscriber) error
-	// Stops the process manager and all running processes within
-	Stop() error
-	// Tells the process manager to associate the deployment request with the given workload ID, and perform any
+type ProcessManager interface { // FIXME-- rename to AgentManager?
+	// Returns a list of agent processes in an implementation-agnostic format
+	// FIXME-- rename to ListRunning()?
```

Comment:
I'd rather not. All processes returned by this function are required to be "running". If there's a process that isn't up yet, the process manager doesn't let anyone know about it.